src/arrayMethodJoin.js

```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const arrCopy = [ ...this ];
+    let concatenatedStr = '';
+
+    if (!arrCopy.length) {
+      return '';
+    }
+
+    for (let i = 0; i < arrCopy.length; i++) {
+      if (arrCopy[i] === undefined || arrCopy[i] === null) {
+        arrCopy[i] = '';
+      }
+    }
+
+    for (let i = 0; i < arrCopy.length - 1; i++) {
+      arrCopy[i] += separator;
+
+      if (separator === null) {
+        arrCopy[i] += 'null';
+      }
+
+      concatenatedStr += arrCopy[i];
+    }
+
+    concatenatedStr += arrCopy[arrCopy.length - 1];
+
+    return concatenatedStr;
```

Comment:
we can simplify this logic using only one loop and without copy the array
think about how to implement it and feel free to ask in the chat